SAP2000_Adapter/CRUD/Read/Results/BarForceSummation.cs

```diff
@@ -0,0 +1,286 @@
+
+                IEnumerable<BarForce> nonSeismicForceAtBar = nonSeismicForces.Where(x => x.ObjectId.ToString() == id);
+
+                //Group by position (do summation at each end)
+                var groupbyposition = forceGroup.GroupBy(x => x.Position);
+
+                foreach (var group in groupbyposition)
+                {
+                    position = group.First().Position;
+                    IEnumerable<BarForce> nonSeismicForceAtPosition = nonSeismicForceAtBar.Where(x => x.Position == position);
+
+                    maxUtilization = group.Max(x => CombineResults(x, barCapacity, gravityDict[id], nonSeismicForceAtPosition, request));
+
+                    result.Add(maxUtilization);
+                }
+            }
+
+            return result;
+        }
+
+        /***************************************************/
+
+        private static AISCSteelUtilisation CombineResults(BarForce timeHistory, Dictionary<string, double> capacity, BarForce gravityDemand, IEnumerable<BarForce> nonSeismicForce, BarForceTimeHistoryRequest request)
+        {
+            BarForce dead = nonSeismicForce.Where(x => x.ResultCase.ToString() == request.NonSeismicCases()[0].ToString()).FirstOrDefault();
+            BarForce live = nonSeismicForce.Where(x => x.ResultCase.ToString() == request.NonSeismicCases()[1].ToString()).FirstOrDefault();
+            BarForce tempPlus = nonSeismicForce.Where(x => x.ResultCase.ToString() == request.NonSeismicCases()[2].ToString()).FirstOrDefault();
+            BarForce tempMinus = nonSeismicForce.Where(x => x.ResultCase.ToString() == request.NonSeismicCases()[3].ToString()).FirstOrDefault();
+
+            double factor = request.Factor;
+            List<BarForce> combos = new List<BarForce>
+            {
+                Add(Add(Multiply(dead, 1.2), Multiply(Subtract(timeHistory, dead), factor)), Add(Multiply(live, 0.2), tempPlus)),
+                Add(Add(Multiply(dead, 1.2), Multiply(Subtract(timeHistory, dead), factor)), Add(Multiply(live, 0.2), tempMinus)),
+                Add(Add(Multiply(dead, 0.9), Multiply(Subtract(timeHistory, dead), factor)), tempPlus),
+                Add(Add(Multiply(dead, 0.9), Multiply(Subtract(timeHistory, dead), factor)), tempMinus)
+            };
+
+            BarForce absMax = Engine.Results.Query.MaxEnvelope(combos);
+            BarForce max = Engine.Results.Query.MinEnvelope(combos);
+            BarForce min = Engine.Results.Query.MinEnvelope(combos);
+
+            double tensionCompressionRatio = Math.Max(max.FX / capacity["T"], min.FX / capacity["P"]);
+            double majorShearRatio = absMax.FZ / capacity["Vz"];
+            double minorShearRatio = absMax.FY / capacity["Vy"];
+            double torsionRatio = absMax.MX / capacity["Mx"];
+            double majorBendingRatio = absMax.MY / capacity["My"];
+            double minorBendingRatio = absMax.MZ / capacity["Mz"];
+
+            double totalRatio = double.NaN;
```

Comment:
@jcparker64 how to combine utilizations?
```suggestion
            double totalRatio = (tensionCompressionRatio + majorBendingRatio + minorBendingRatio) + Math.Pow(majorShearRatio + minorShearRatio + torsionRatio,2);
```

Reply:
Per AISC 360-16 Chapter H

Reply:
It's conditional per H3.1 and H3.2. If torsional DCR is <= 0.2 we just use Eq. H1-1a/H1-1b
Otherwise, we use Eq. H3-6

Reply:
Got it- see latest commit; H1-1a and b